components/brave_ads/core/browser/internals/ads_internals_handler.cc

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_ads/core/browser/internals/ads_internals_handler.h"
+
+#include <utility>
+
+#include "base/check.h"
+#include "base/functional/bind.h"
+#include "base/json/json_writer.h"
+#include "brave/components/brave_ads/core/browser/service/ads_service.h"
+#include "brave/components/brave_rewards/common/pref_names.h"
+#include "brave/components/services/bat_ads/public/interfaces/bat_ads.mojom.h"
+#include "components/prefs/pref_service.h"
+
+AdsInternalsHandler::AdsInternalsHandler(brave_ads::AdsService* ads_service,
+                                         PrefService* prefs)
+    : ads_service_(ads_service), prefs_(prefs) {
+  CHECK(ads_service_);
+  CHECK(prefs_);
```

Comment:
Would it be possible to have these values passed in by reference since they cannot be null?

Reply:
`ads_service_` can be null for Private window, so I removed the `CHECK(ads_service_);`. Will update the test plan.